### PR TITLE
[FIX] Align EEGTCNet, SSTDPN & ATCNet with original source implementations

### PR DESCRIPTION
**Pull request by bruAristimunha**
## Summary

Reproducing several braindecode models against their authors' original implementations surfaced divergences that hurt source-faithful reproduction. This PR patches three models, closing #1059, #1060, and #1061 (root-caused in [bruAristimunha/eeg-replicatibility#3](https://github.com/bruAristimunha/eeg-replicatibility/issues/3)).

### EEGTCNet (#1060)
- **Separate dropout rates** via new `drop_prob_eeg` / `drop_prob_tcn` (default `None` → fall back to `drop_prob`), enabling the paper's `p_eeg=0.2`, `p_tcn=0.3`. *Backward-compatible.*
- **Source-faithful TCN BatchNorm**: the Keras source places a `BatchNormalization` after every TCN convolution (`momentum=0.99` → torch `0.01`, `eps=1e-3`) and uses a biased residual downsample; braindecode omitted both. Added behind `tcn_batch_norm: bool = True`. ⚠️ **Behavior change**: this alters the default architecture and parameter count (4.20K → 4.30K trainable, closer to the paper's reported 4.27K). Pass `tcn_batch_norm=False` to recover the previous architecture (e.g. to load checkpoints trained before this change).

### SSTDPN (#1059)
- **Bug fix**: `proto_sep` was renormalized along `dim=1` (across classes). The paper/source constrains each class-row prototype (`‖s_i‖ ≤ S`), i.e. `dim=0`. Fixed.
- **`proto_cpt_std` default `0.01` → `1.0`** to match the source `torch.randn` initialization. `proto_cpt` is not used in `forward`, so this is init-only.

### ATCNet (#1061)
- **Opt-in `conv_max_norm_const`** (default `None`): applies the official max-norm constraint (`0.6`) to the conv/TCN kernels via a `MaxNormParametrize` weight parametrization. *Backward-compatible.*
- **`source_optimizer_param_groups()`**: returns the official `L2` weight-decay groups (conv/TCN `0.009`, dense `0.5`, everything else `0.0`) for use with any `torch.optim` optimizer.

## Backward compatibility
All additions are opt-in or fall back to prior behavior **except** the EEGTCNet TCN BatchNorm, which is documented under "API and behavior changes" in `whats_new.rst` with the `tcn_batch_norm=False` escape hatch.

## Tests
- New regression tests for each fix (TCN BatchNorm presence + legacy recovery, separate dropout routing, `proto_sep` per-class-row constraint, `proto_cpt_std` default, conv/TCN max-norm clamp, optimizer param-group coverage including the parametrized path).
- Updated `test_parameters_EEGTCNet` to the new 4.3K count.
- All model contract tests pass locally (config/HF round-trip, integration, batch-size-1 train mode, export, `jit.script`).

## Out of scope
The non-model items from eeg-replicatibility#3 (Deep4Net cropped config, CTNet/TIDNet evaluation protocol, SSTDPN dual-optimizer training loop) are training-pipeline concerns rather than model code, and are not addressed here.

### CODE REVIEW

File: braindecode/models/atcnet.py [hunk 1]
```diff
@@ -458,6 +479,90 @@ def forward(self, X: torch.Tensor) -> torch.Tensor:
...
+        dense_weight_decay: float = 0.5,
+    ) -> list[dict]:
+        r"""Source-faithful optimizer parameter groups for ATCNet.
+
+        The official Keras implementation applies ``L2`` weight decay only to the
+        convolution / TCN kernels (``conv_weightDecay = 0.009``) and to the final
+        dense layer (``dense_weightDecay = 0.5``); biases, BatchNorm and attention
+        weights are left undecayed. PyTorch's ``weight_decay`` plays the role of
+        Keras' ``L2`` penalty, so passing these groups to an optimizer reproduces
+        the source regularization.
```
[the remark this answers is not in the document]
Comment (bruAristimunha) [reply braindecode/models/atcnet.py]: Good catch — clarified in b9280a73: the docstring now states that `weight_decay` reproduces the Keras `L2` penalty only for coupled-decay optimizers (`torch.optim.SGD`/`Adam`, as in the official code), and that `torch.optim.AdamW` is *not* equivalent.

File: braindecode/models/eegtcnet.py [hunk 2]
```diff
@@ -277,50 +302,58 @@ def __init__(
...
+                dilation=dilation,
+                padding=padding,
+                bias=False,
+            ),
+            Chomp1d(padding),
+        ]
+        if self.batch_norm:
+            # Keras BatchNormalization defaults: momentum=0.99 (-> 0.01 in
+            # PyTorch convention) and epsilon=1e-3.
+            unit.append(nn.BatchNorm1d(self.filters, momentum=0.01, eps=1e-3))
```
[the remark this answers is not in the document]
Comment (bruAristimunha) [reply braindecode/models/eegtcnet.py:355]: Confirmed and fixed in b9280a73. Reproduced the crash (`EEGTCNet(n_times=64)`, batch size 1, train mode → `Expected more than 1 value per channel`); `forward` is now wrapped with `_disable_batch_norm_training_if_batch_size_one` (the same guard used by ContraWR/EEGMiner/USleep), and a regression test at `n_times=64` was added.
